saleor/graphql/core/connection.py

```diff
@@ -1,7 +1,118 @@
+from typing import Any, Dict
+
 import graphene
+from django.db.models.query import QuerySet
 from graphene import Field, List, NonNull, ObjectType, String
 from graphene.relay.connection import Connection
 from graphene_django_optimizer.types import OptimizedDjangoObjectType
+from graphql_relay.connection.connectiontypes import Edge, PageInfo
+
+ConnectionArguments = Dict[str, Any]
+
+
+def connection_args_validation(args):
+    first = args.get("first")
+    last = args.get("last")
+
+    if isinstance(first, int) and isinstance(last, int):
+        raise ValueError("Argument 'last' cannot be combined with 'first'.")
+    if isinstance(first, int) and args.get("before"):
+        raise ValueError("Argument 'first' cannot be combined with 'before'.")
+    if isinstance(last, int) and args.get("after"):
+        raise ValueError("Argument 'last' cannot be combined with 'after'.")
+    if isinstance(first, int) and first < 0 or first and not isinstance(first, int):
+        raise ValueError("Argument 'first' must be a non-negative integer.")
+    if isinstance(last, int) and last < 0 or last and not isinstance(last, int):
+        raise ValueError("Argument 'last' must be a non-negative integer.")
+
```

Comment:
I think that `connection_args_validation` could be simplified. Firstly we should check if `last` and `first` are non-negative integer and then `isinstance()` check is not needed anymore. Like:
```suggestion
    for arg in ["first", "last"]:
        arg_value = args.get(arg)
        if isinstance(arg_value, int) and arg_value < 0 or arg_value and not isinstance(arg_value, int):
            raise ValueError(f"Argument '{arg}' must be a non-negative integer.")
    for arg1, arg2 in [("first", "last"), ("first", "before"), ("last", "after")]:
        if args.get(arg1) and args.get(arg2):
            raise ValueError(f"Argument '{arg1}' cannot be combined with '{arg2}'.")
```